components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2450,6 +2492,196 @@ public void updateGroup(Group oldGroup, Group newGroup) throws CharonException {
+                if (memberOperation.getValues() instanceof Map) {
+                    Map<String, String> memberObject = (Map<String, String>) memberOperation.getValues();
+                    prepareAddedRemovedMemberLists(addedMembers, deletedMembers, newlyAddedMemberIds, memberOperation,
+                            memberObject);
+                } else if (memberOperation.getValues() instanceof List) {
+                    List<Map<String, String>> memberOperationValues =
+                            (List<Map<String, String>>) memberOperation.getValues();
+                    for (Map<String, String> memberObject : memberOperationValues) {
+                        prepareAddedRemovedMemberLists(addedMembers, deletedMembers, newlyAddedMemberIds,
+                                memberOperation, memberObject);
+                    }
+                }
+            }
+
+            String userStoreDomainForGroup = IdentityUtil.extractDomainFromName(newGroupName);
+            Set<String> temporaryMembers = new HashSet<>();
+
+            if (isNotInternalOrApplicationGroup(userStoreDomainForGroup) && (!addedMembers.isEmpty()
+                    || !deletedMembers.isEmpty())) {
+                for (String member : addedMembers) {
+                    String username = UserCoreUtil.addDomainToName(UserCoreUtil.removeDomainFromName(member),
+                            userStoreDomainForGroup);
+                    temporaryMembers.add(username);
+                }
+
+                addedMembers.clear();
+                addedMembers.addAll(temporaryMembers);
+                temporaryMembers.clear();
+
+                for (String member : deletedMembers) {
+                    String username = UserCoreUtil.addDomainToName(UserCoreUtil.removeDomainFromName(member),
+                            userStoreDomainForGroup);
+                    temporaryMembers.add(username);
+                }
+
+                deletedMembers.clear();
+                deletedMembers.addAll(temporaryMembers);
+            }
+
+            // Check for deleted members
+            Set<String> deletedMemberIdsFromUserstore =
+                    getMemberValuesFromUserstore(deletedMembers, userStoreDomainForGroup, newGroupName);
+
+            // Check for added members
+            Set<String> addedMemberIdsFromUserstore =
+                    getMemberValuesFromUserstore(addedMembers, userStoreDomainForGroup, newGroupName);
+
+            // Validate the memberIds sent in the update request against the Ids retrieved from the user store.
+            if (isNotEmpty(addedMembers)) {
+                validateUserIds(addedMemberIdsFromUserstore, newlyAddedMemberIds);
+            }
+
+            /*
+            Set thread local property to signal the downstream SCIMUserOperationListener
+            about the provisioning route.
+            */
+            SCIMCommonUtils.setThreadLocalIsManagedThroughSCIMEP(true);
+
+             /*
```

Comment:
Shall we fix alignment here?

Reply:
Resolved in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/280/commits/36add81b74fe1bfc479adf2b8217e00af089b8f7